libs/fetcher.js

```diff
@@ -28,6 +28,25 @@ function parseParamValues (params) {
     }, {});
 }
 
+function getErrorResponse(err) {
+    var statusCode = err.statusCode || 400;
+    var output = {
+        message: 'request failed'
+    };
+
+    if (typeof err.output !== 'undefined') {
+        output = err.output;
+    } else if (err.message) {
```

Comment:
> is this for backwards compatibility?

Yes.

> could we just remove the else and make the `if (err.message)` its own block?


No, that would tamper with your special `.output` if you have thrown an error with a message, which would lead to some confusing outputs & defeats the purpose of being able to customise the output.